### ISSUE TITLE
Migration of filesystems encrypted with PRE-1.4.4.0 version to the current 1.4.4.0 version

### ISSUE THREAD

**Issue by san-r** [bailey27/cppcryptfs#164]
cppcryptfs version 1.4.4.0 implements the "deterministic names" feature which makes diriv files unnecessary. This is very significant for me.

I have several folders encrypted with previous cppcryptfs version which have diriv files. My question is - Is there a faster way to convert those encrypted folders to the newer version (without diriv files) other than mounting both older and newer filesystems and then copying all files from older location to newer location?

**Comment by bailey27** [comment]
Currently, the only way to migrate is to copy everything.

**Comment by loungebob** [comment]
no. nooo. nooooooo. I was afraid something like this was going to happen. I need a migration path for multiple Petabytes of gocryptfs volumes. copying is just not an option. I'll go and lobby jakob.

**Comment by san-r** [comment]
I was hoping that the migration process should involve just -

- deleting the diriv files
- updating the config file

without touching the encrypted data files.

**Comment by loungebob** [comment]
me too.

**Comment by ccchan234** [comment]
it would be nice if implement of "longnamemax" didn't need to COPY files to change container too.

I got around 30TB files on gdrive.

thanks

**Comment by yfdyh000** [comment]
I think this is doable by retrofitting the https://github.com/slackner/gocryptfs-inspect. But I don't have enough time to do this. In any case, I recommend whoever implements it to keep shadows, logs and tests intact in case any files are missing or crash halfway through.

In short, full rename log, first rename old diriv files to .bak and backup them, and let user check number / names of files inside volume.

**Comment by yfdyh000** [comment]
> #metoo
> 
> it would be nice if implement of "longnamemax" didn't need to COPY files to change container too.
> 
> I got around 30TB files on gdrive.
> 
> thanks

Upvoted this, I've encountered sync failures due to max paths exceeding sync software limits, from long base path, no way to change after the volume is created, I had to shorten the naming of existing folders to the shortest possible.
